src/locale/vi/_lib/localize/index.js

```diff
@@ -1,53 +1,113 @@
 import buildLocalizeFn from '../../../_lib/buildLocalizeFn/index.js'
-import buildLocalizeArrayFn from '../../../_lib/buildLocalizeArrayFn/index.js'
 
-// Vietnammese locale reference: http://www.localeplanet.com/icu/vi-VN/index.html
-// Capitalization reference: http://hcmup.edu.vn/index.php?option=com_content&view=article&id=4106%3Avit-hoa-trong-vn-bn-hanh-chinh&catid=2345%3Atham-kho&Itemid=4103&lang=vi&site=134
-var weekdayValues = {
-  narrow: ['CN', 'T2', 'T3', 'T4', 'T5', 'T6', 'T7'],
-  short: ['CN', 'thứ 2', 'thứ 3', 'thứ 4', 'thứ 5', 'thứ 6', 'thứ 7'],
-  long: ['Chủ Nhật', 'thứ Hai', 'thứ Ba', 'thứ Tư', 'thứ Năm', 'thứ Sáu', 'thứ Bảy']
+var eraValues = {
+  narrow: ['TCN', 'SCN'],
+  abbreviated: ['trước CN', 'sau CN'],
+  wide: ['trước Công Nguyên', 'sau Công Nguyên']
+}
+
+var quarterValues = {
+  narrow: ['1', '2', '3', '4'],
+  abbreviated: ['Q1', 'Q2', 'Q3', 'Q4'],
+  wide: ['quý 1', 'quý 2', 'quý 3', 'quý 4']
+}
+var formattingQuarterValues = {
+  narrow: ['1', '2', '3', '4'],
+  abbreviated: ['Q1', 'Q2', 'Q3', 'Q4'],
+  wide: ['Quý 1', 'Quý 2', 'Quý 3', 'Quý 4']
 }
 
+// Note: in English, the names of days of the week and months are capitalized.
+// If you are making a new locale based on this one, check if the same is true for the language you're working on.
+// Generally, formatted dates should look like they are in the middle of a sentence,
+// e.g. in Spanish language the weekdays and months should be in the lowercase.
 var monthValues = {
+  narrow: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'],
+  short: ['Thg 1', 'Thg 2', 'Thg 3', 'Thg 4', 'Thg 5', 'Thg 6', 'Thg 7', 'Thg 8', 'Thg 9', 'Thg 10', 'Thg 11', 'Thg 12'],
```

Comment:
I don't think `monthValues` has **short** form as per i18n documentation and the format notation for Months are: M (number), Mo (ordinal number), MM (narrow), MMM (abbreviated), MMMM (wide).